cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -343,6 +343,7 @@ IndexMap::IndexMap(MPI_Comm comm, std::int32_t local_size)
   _comm_owner_to_ghost = dolfinx::MPI::Comm(comm0, false);
   _comm_ghost_to_owner = dolfinx::MPI::Comm(comm1, false);
   _comm_symmetric = dolfinx::MPI::Comm(comm2, false);
+  _shared_indices = std::make_unique<graph::AdjacencyList<std::int32_t>>(0);
```

Comment:
I don't follow what's happening here - why is `_shared_indices` being initialised? Are there no 'shared indices'?

Reply:
This initialization is in the non-overlapping index-map constructor, so there are no 'shared indices', and no ghosts.
It's being initialized to avoid the need for checking if the unique_ptr is "empty".